### ISSUE TITLE
Move to PyPI trusted publishing

### ISSUE THREAD

**Issue by kafitzgerald** [NCAR/geocat-datafiles#69]
It looks like we still have a token in here for the PyPI workflow and could use to change that to trusted publishing to minimize the use of tokens where no longer needed.  This should look similar to the [corresponding change over on geocat-comp](https://github.com/NCAR/geocat-comp/issues/533).

**Comment by kafitzgerald** [comment]
I think we'd discussed waiting on this for #71, since that will address a number of build related deprecations

**Comment by kafitzgerald** [comment]
This should be somewhat less blocked now.

I think it's still worth trying a release first with the current authentication method though and then updating after that.
